### PR TITLE
Handle DOS files in SynthonSpaceSearch

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
No issue but Igor from Enamine mentioned it was a problem.

#### What does this implement/fix? Explain your changes.
Uses the RDKit getLine() rather than std::get line() which handles CRLF.

#### Any other comments?
I think the test is problematic as the newlines are fixed on git checkout, at least on my Ubuntu development area.

### CODE REVIEW

**Comment by bp-kelley** [comment]
In .gitattribute you can control line endings:

path/to/my/file eol=crlf

**Review by greglandrum — CHANGES_REQUESTED** [review]
The code changes look good, but please use a shorter test file. We don't need to add another long-running test for this one.

File: Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp [hunk 1]
```diff
@@ -550,4 +550,14 @@ TEST_CASE("Greg Space Failure") {
...
   CHECK(results.getHitMolecules().size() == 1);
+}
+
+TEST_CASE("DOS File") {
+  REQUIRE(rdbase);
+  std::string fName(rdbase);
+  std::string libName = fName + "/Code/GraphMol/SynthonSpaceSearch/data/amide_space_dos.txt";
+  SynthonSpace synthonspace;
+  synthonspace.readTextFile(libName);
+  std::cout << "num prods : " << synthonspace.getNumProducts() << std::endl;
```
Comment (greglandrum) [inline Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp:561]: Last suggested change, please remove this.
```suggestion
```

Comment (greglandrum) [reply Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp:561]: actually, I can do that

**Review by greglandrum — APPROVED** [review]
LGTM
will merge after the CI builds pass

**Pull request merged**